### PR TITLE
Make teams available on forwardings

### LINKED ISSUE #3888: Teams stehen nicht zur Auswahl bei Weiterleitungen
Bei der Erstellung bzw. dem Neuzuweisen einer Weiterleitung stehen die Teams fälschlicherweise nicht zur Verfügung.

### PR DESCRIPTION
**Pull request by phgross**
And support teams also when reassigning task/forwardings, but only when the task/forwarding is in an open state, otherwise the User is informed via message.

bildschirmfoto 2018-01-24 um 18 10 13

Closes #3888 

Was a missing part of the OGIP 24 Teamaufgaben

### CODE REVIEW

File: opengever/inbox/tests/test_forwarding.py [hunk 1]
```diff
@@ -74,3 +74,22 @@ def test_forwarding_edit_form_does_not_render_empty_fieldset_additional(
...
+        browser.fill({'Title': u'Test forwarding'})
+
+        # Fill Responible manually
+        form = browser.find_form_by_field('Responsible')
+        form.find_widget('Responsible').fill('team:1')
+        browser.find('Save').click()
+
+        forwarding = browser.context.objectValues()[-1]
+
+        self.assertEqual('team:1', forwarding.responsible)
```
Comment (deiferni) [inline opengever/inbox/tests/test_forwarding.py]: could this test somehow involve the team title?

File: opengever/task/browser/assign.py [hunk 2]
```diff
@@ -81,13 +81,14 @@ def handle_assign(self, action):
...
+                    self.context.absolute_url())
 
-            self.reassing_task(**data)
+            self.reassign_task(**data)
 
-        return self.request.RESPONSE.redirect('.')
+        return self.request.RESPONSE.redirect(self.context.absolute_url())
 
-    def reassing_task(self, **kwargs):
+    def reassign_task(self, **kwargs):
```
Comment (deiferni) [inline opengever/task/browser/assign.py]: 🎶 🎤

File: opengever/task/tests/test_assign.py [hunk 3]
```diff
@@ -5,110 +5,109 @@
...
+        browser.click_on('Assign')
+
     @browsing
     def test_responsible_client_and_transition_field_is_hidden(self, browser):
-        browser.login().open(self.task, view='assign-task')
+        self.login(self.regular_user, browser=browser)
 
+        browser.open(self.task, view='assign-task')
         self.assertEquals(None, browser.find('Responsible Client'))
+        self.assertEquals(None, browser.find('Transition'))
```
Comment (deiferni) [inline opengever/task/tests/test_assign.py]: `assertIsNone`

File: opengever/inbox/tests/test_forwarding.py [hunk 4]
```diff
@@ -93,3 +93,16 @@ def test_teams_are_available_as_responsible(self, browser):
...
         self.assertEqual('team:1', forwarding.responsible)
         self.assertEqual(u'Test forwarding', forwarding.title)
+
+    @browsing
+    def test_forwarding_can_reassigned_to_a_team(self, browser):
+        self.login(self.secretariat_user, browser=browser)
+        browser.open(self.inbox_forwarding)
+        browser.click_on('forwarding-transition-reassign')
+
+        # Fill Responible manually
```
Comment (deiferni) [inline opengever/inbox/tests/test_forwarding.py]: Respon**s**ible

File: opengever/task/browser/assign.py [hunk 5]
```diff
@@ -49,6 +55,29 @@ class IAssignSchema(Schema):
...
 
 
+class NoTeamsInProgressStateValidator(validator.SimpleFieldValidator):
+    """Teams are only allowed if the task/forwarding is in open state.
+    """
+
+    def validate(self, value):
+        if ActorLookup(value).is_team():
+            if api.content.get_state(self.context) not in [
+                    'task-state-open', 'forwarding-state-open']:
```
Comment (deiferni) [inline opengever/task/browser/assign.py]: you should use the `is_open` method on task here (or implement if not available).

File: opengever/task/locales/de/LC_MESSAGES/opengever.task.po [hunk 6]
```diff
@@ -211,6 +211,11 @@ msgstr "Verantwortliche"
 msgid "error_checked_out_document"
 msgstr "Die folgenden Dokumente (${title}) müssen noch eingecheckt werden."
 
+#. Default: "Team responsibles are only allowed if the task or forwarding is open."
+#: ./opengever/task/browser/assign.py
+msgid "error_no_team_responsible_in_progress_state"
+msgstr "Teams Auftragnehmer sind nur bei Aufgaben oder Weiterleitungen im Status offen erlaubt."
```
Comment (deiferni) [inline opengever/task/locales/de/LC_MESSAGES/opengever.task.po]: deutsch 😉

File: opengever/task/browser/assign.py (hunk 5 above)
Comment (Rotonen) [reply opengever/task/browser/assign.py]: Please do that for all the other workflow states as well when you go for that.

**Comment by phgross** [comment]
@deiferni updated ...

**Pull request merged**